src/Managers/JoinedRoomsManager.ts

```diff
@@ -0,0 +1,63 @@
+export default class JoinedRoomsManager {
+    // Map of room ID to set of our bot user IDs in the room
```

Comment:
I think this class logic could b be merged into `BotUsersManager`? It seems quite intertwined.